app/grandchallenge/evaluation/models.py

```diff
@@ -1670,6 +1670,20 @@ def create_evaluation(self, *, additional_inputs):
                 status=Evaluation.VALIDATING_INPUTS,
             )
 
+        if not self.has_matching_algorithm_interfaces:
+            evaluation.update_status(
+                status=Evaluation.CANCELLED,
+                error_message="The algorithm interfaces do not match those defined for the phase.",
+            )
+
+        if self.has_blocking_algorithm_jobs:
+            evaluation.update_status(
+                status=Evaluation.CANCELLED,
+                error_message="There are non-successful jobs for this submission. "
+                "These need to be handled first before you can "
+                "re-evaluate. Please contact support.",
+            )
+
```

Comment:
I have opted to create the evaluation and mark it as failed for more direct feedback to the admins. The alternative would be to send a notification like we do for missing methods, but I feel like those just get lost, and we (siet admins) wouldn't get those.